core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -6661,6 +6661,11 @@ static class ExtendedExpander extends Expander {
           SqlIdentifier sid = (SqlIdentifier) expr;
           final SqlIdentifier fqId = getScope().fullyQualify(sid).identifier;
           expr = expandDynamicStar(sid, fqId);
+        } else {
+          requireNonNull(expr, "expr");
```

Comment:
Are we sure `expr` will always be non-null here?

Reply:
Thanks ruben, I dove into SqlNodeList which is type of selectList and found its item could be nullable, but I couldn't found any possible nullable selectItem in parser.jj, so I think selectItem must be non-nullable, do you think so?